mlflow/tensorflow/__init__.py

```diff
@@ -971,44 +978,60 @@ def _extract_n_steps(input_example_n_steps):
                             ]
 
                         else:
-                            warnings.warn(
-                                "Tensorflow keras autologging only "
-                                "supports input types of: numpy.ndarray, "
-                                "dict(<key> -> numpy.ndarray), tensorflow.data.Dataset, "
-                                "or tensorflow.keras.utils.Sequence"
```

Comment:
Warning and returning `None` produces a strange message when this method is called by `resolve_input_example_and_signature`; something to the effect of "cannot concatenate NoneType and String" because `input_example_slice is `None`. Instead of returning `None`, we should throw. `resolve_input_example_and_signature` handles graceful incorporation of this error message into a warning message.